services/file-retriever/src/services/dsnFetcher.ts

```diff
@@ -0,0 +1,99 @@
+import { FileResponse } from '../models/file.js'
+import {
+  blake3HashFromCid,
+  stringToCid,
+  decodeNode,
+  MetadataType,
+  cidToString,
+} from '@autonomys/auto-dag-data'
+import { PBNode } from '@ipld/dag-pb'
+import { env } from '../utils/env.js'
+import { HttpError } from '../http/middlewares/error.js'
+import { safeIPLDDecode } from '../utils/dagData.js'
+import mime from 'mime-types'
+
+const SUBSPACE_GATEWAY_URL = env('SUBSPACE_GATEWAY_URL')
+const MAX_SIMULTANEOUS_FETCHES = 10
+
+const fetchNode = async (cid: string): Promise<PBNode> => {
+  const objectMappingHash = blake3HashFromCid(stringToCid(cid))
+  const response = await fetch(
+    `${SUBSPACE_GATEWAY_URL}/objects/${objectMappingHash}`,
+  )
+  if (!response.ok) {
+    throw new HttpError(500, 'Internal server error: Failed to fetch node')
+  }
+
+  const blob = await response.arrayBuffer()
+
+  return decodeNode(blob)
```

Comment:
I wonder, what does it do?  The blob should be a binary file chunk. What decoding do we need?

Reply:
It's an [IPLD node (in protobuf version)](https://ipld.io/specs/codecs/dag-pb/spec/) not just a file chunk so we should decode the IPLD node first.